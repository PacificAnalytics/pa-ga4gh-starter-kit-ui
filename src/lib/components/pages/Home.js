import '@fontsource/roboto';
import React from 'react';
import {
  Link
} from "react-router-dom";
import { 
  Container, 
  Button, 
  Typography,
  Breadcrumbs
 } from '@material-ui/core';
 import { scrollToTop } from '../../functions/common';
import PageContainer from '../common/layout/PageContainer';
import BackButton from '../common/button/BackButton';
import BreadcrumbTrail from '../common/navigation/BreadcrumbTrail';

const Home = props => {

    return(
        <PageContainer>
            <BreadcrumbTrail trail={props.trail} />

            <Typography variant="h1" gutterBottom>Welcome to the GA4GH Starter Kit</Typography>
            <Button
                variant="outlined"
                color="secondary"
            >
                <Link to='/services'>
                    <Typography variant="button">Get Started</Typography>
                </Link>
            </Button>
        </PageContainer>
    );
}

export default Home;
