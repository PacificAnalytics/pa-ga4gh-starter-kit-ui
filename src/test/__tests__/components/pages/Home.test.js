import React from 'react';
import {MemoryRouter, Route} from 'react-router-dom';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import Home from '../../../../lib/components/pages/Home';

test('the correct Home page elements are displayed', () => {
    let testLocation;
    let {container} = render(
        <MemoryRouter>
            <Home/>
            <Route 
                path="*"
                render={({ location }) => {
                testLocation = location;
                return null;
                }}
            />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByRole('heading', {level: 1})).toHaveTextContent('Welcome to the GA4GH Starter Kit');
    expect(screen.getByRole('heading', {level: 3})).toHaveTextContent('Get Started');
    expect(screen.getByText('Click the buttons below to start using one of the GA4GH Starter Kits')).toBeInTheDocument();
    const drsStarterKitButton = screen.getByRole('button');
    expect(drsStarterKitButton).toHaveTextContent('DRS Starter Kit');
    userEvent.click(drsStarterKitButton);
    expect(testLocation.pathname).toBe('/drs');
})