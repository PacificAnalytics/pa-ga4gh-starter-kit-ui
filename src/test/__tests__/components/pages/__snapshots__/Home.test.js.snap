// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the correct Home page elements are displayed 1`] = `
<div
  align="center"
>
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom"
    >
      Welcome to the GA4GH Starter Kit
    </h1>
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
    >
      Get Started
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Click the buttons below to start using one of the GA4GH Starter Kits
    </p>
    <div>
      <nav>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/drs"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              DRS Starter Kit
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </nav>
    </div>
  </div>
</div>
`;
